### DISCUSSION TITLE
Dynamic mongo database selection within same cluster

### DISCUSSION THREAD

**Discussion by rohit-genvis** [prisma/prisma#17805] [Q&A]
We have a use case where we are running multiple tenant environments (multiple databases) running in the same Mongo cluster. All databases share the same schema. The database used for action will be decided based on an id in the API request. What is the best way to handle this using Prisma? Creating multiple dynamic PrismaClients is one solution, any alternative recommended approach? The approach discussed in https://github.com/Errorname/prisma-multi-tenant can be used, but this will not allow connection sharing which is supported using Mongo native libs

**Comment by nurul3101** [comment]
Hello @rohit-genvis 👋 

I would say you can create multiple PrismaClients similar to this [example](https://github.com/prisma/prisma/issues/2443#issuecomment-630679118), and based on the request id, determine with PrismaClient should be used. That would be the most straightforward way to achieve your use case.

**Reply by rohit-genvis** [reply]
thanks @nurul3101 , issue with this approach is underlining connection pool is not shared

**Reply by janpio** [reply]
Prisma can currently only handle 1 database per connection pool.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
